docs/content/latest/explore/ysql-language-features/data-manipulation.md

```diff
@@ -0,0 +1,343 @@
+
+```sql
+CREATE TABLE employees (
+    employee_no integer,
+    name text,
+    department text
+  		DEFERRABLE INITIALLY DEFERRED
+);
+```
+
+You can change this behavior by altering a constraint for a table or column, as shown in the following example:
+
+```sql
+ALTER TABLE employees 
+ALTER CONSTRAINT employee_no
+  DEFERRABLE INITIALLY DEFERRED;
+```
+
+Note that the `NOT NULL` constraint cannot be used with the `SET CONSTRAINTS` statement.
+
+## Obtaining Modified Data
+
+The `RETURNING` clause allows you to obtain data in real time from the rows that you modified using the `INSERT`, `UPDATE`, and `DELETE` statements.
+
+The `RETURNING` clause can contain either column names of its parent statement's target table or value expressions using these columns. To select all columns, in order, of the target table, use `RETURNING *`.
+
+When you use the `RETURNING` clause within the  `INSERT` statement, you are obtaining data of the row as it was inserted. This is useful when dealing with computed default values or with `INSERT ... SELECT` .
+
+When using the `RETURNING` clause within the  `UPDATE` statement, the data you are obtaining from `RETURNING` represents the new content of the modified row, as the following example demonstrates:
+
+```sql
+UPDATE employees SET employee_no = employee_no + 1
+	WHERE employee_no = 1
+	RETURNING name, employee_no AS new_employee_no;  
+```
+
+In cases of using the `RETURNING` clause within the  `DELETE` statement, you are obtaining the content of the deleted row, as shown the following example:
+
+```sql
+DELETE FROM employees WHERE department = 'Sales' RETURNING *;  
+```
+
+## Auto-Incrementing Column Values
+
+Using a special kind of database object called sequence, you can generate unique identifiers by auto-incrementing the numeric identifier of each preceding row. In most cases, you would use sequences to auto-generate primary keys.
+
+Typically, you add sequences using the `serial` pseudotype that creates a new sequence object and sets the default value for the column to the next value produced by the sequence. 
+
+When a sequence generates values, it adds a `NOT NULL` constraint to the column. 
+
+The sequence is automatically removed if the  `serial` column is removed.
+
+You can create both a new table and a new sequence generator at the same time, as follows: 
+
+```sql
+CREATE TABLE employees (
+    employee_no serial,
+    name text,
+    department text,
+);
+```
+
+You may also choose to assign auto-incremented sequence values to new rows created via the `INSERT`  statement. To instruct `INSERT` to take the default value for a column, you can omit this column from the `INSERT` column list, as shown in the following example:
+
+```sql
+INSERT INTO employees (name, department) VALUES ('John Smith', 'Sales');  
+```
+
```

Comment:
another great place for a `SELECT` and the expected output.

Reply:
Thanks.